pkg/bpf/map_linux.go

```diff
@@ -671,7 +671,11 @@ func (m *Map) DumpReliablyWithCallback(cb DumpCallback, stats *DumpStats) error
 				// Depending on exactly when currentKey was deleted from the
 				// map, nextKey may be the actual key element after the deleted
 				// one, or the first element in the map.
-				currentKey = nextKey
+
+				// We can't simply assign currentKey to equal nextKey because
+				// that will result in currentKey and nextKey pointing to the
+				// same memory (which causes unexpected behavior when calling ebpf.NextKey).
+				m.NextKey(currentKey, currentKey)
```

Comment:
What's the behavior of `m.NextKey` if you call it with a key that got deleted? Will it set `currentKey` to the first key in the map? And if so couldn't we just do `m.NextKey(nil, currentKey)`? I think the comment could explain this a bit more, in addition to explaining why `currentKey = nextKey` isn't working.

And shouldn't we check the returned error?

Reply:
Looking at this again, I think I was overthinking this.  I have a simpler solution that avoids the extra NetKey call now.